### PR TITLE
MISUV-8052-investigate-httpClientV2

### PR DESCRIPTION
**Pull request by simondrugan16**
investigation for implementing http client v2 + separate HTTP reads

### CODE REVIEW

File: test/connectors/ChargeHistoryConnectorSpec.scala [hunk 1]
```diff
@@ -18,29 +18,31 @@ package connectors
...
+import testConstants.ChargeHistoryTestConstants.{testChargeHistoryErrorModelParsing, testValidChargeHistoryModel}
 import testUtils.TestSupport
-import uk.gov.hmrc.http.HttpResponse
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
+import java.time.LocalDate
 import scala.concurrent.Future
 
-class ChargeHistoryConnectorSpec extends TestSupport with MockHttp with MockAuditingService {
+class ChargeHistoryConnectorSpec extends TestSupport with MockHttp with MockHttpV2 with MockAuditingService {
```
Comment (avelx) [inline test/connectors/ChargeHistoryConnectorSpec.scala:34]: I think we don't need MockHttp to be exposed in the scope of this spec? as we can use only one or anther ?

**Comment by owen-gilfellon** [comment]
I was hoping `AResponseReads` might be useful in cutting down on boiler plate when implementing the HttpReads, as per CreditsModel 🙂

`object CreditsModel {

  implicit val format: OFormat[CreditsModel] = Json.format[CreditsModel ]

  implicit val reads: CreditsResponseReads = new CreditsResponseReads
  class CreditsResponseReads extends AResponseReads[CreditsModel] {
    implicit val format: Format[CreditsModel] = CreditsModel.format
  }
}`

**Comment by simondrugan16** [comment]
> I was hoping `AResponseReads` might be useful in cutting down on boiler plate when implementing the HttpReads, as per CreditsModel 🙂
> 
> ```
> object CreditsModel {
> 
>   implicit val format: OFormat[CreditsModel] = Json.format[CreditsModel ]
> 
>   implicit val reads: CreditsResponseReads = new CreditsResponseReads
> 
>   class CreditsResponseReads extends AResponseReads[CreditsModel] {
>     implicit val format: Format[CreditsModel] = CreditsModel.format
>   }
> }
> ```

I like the idea of using AResponseReads but I didn't really want to mess with the logic since the current read logic is a bit different to AResponseReads (e.g: current logic treats NOT_FOUND and FORBIDDEN in the same way 😵‍💫 ). I think it would be a good idea to revisit and simplify these connectors in the future, but probably outside the scope of this ticket? What do you think

**Comment by owen-gilfellon** [comment]
> I like the idea of using AResponseReads but I didn't really want to mess with the logic since the current read logic is a bit different to AResponseReads (e.g: current logic treats NOT_FOUND and FORBIDDEN in the same way 😵‍💫 ). I think it would be a good idea to revisit and simplify these connectors in the future, but probably outside the scope of this ticket? What do you think

Yeah, I was wondering about that - it returns an empty model if FORBIDDEN, seemed strange! Fair enough, if it's going to cause issues

**Review by avelx — APPROVED** [review]

**Pull request merged**